source/Ship.h

```diff
@@ -448,6 +456,8 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Characteristics of this particular ship:
 	std::string name;
 	bool canBeCarried = false;
+	// The bay this ship docks too, if any.
+	Bay *reservedBay = nullptr;
```

Comment:
The parent-ship is handled by a weak-pointer. Is it safe to use a regular pointer here?

Reply:
Now that I think about it, no. Because it points to a `std::vector` which if any bays get added later (due to an event) will invalidate any pointers. I'll change this to a `std::list`.

We can't use a `weak_ptr` here because the bays aren't `shared_ptr`.